src/manager/node.c

```diff
@@ -724,6 +724,20 @@ static int node_disconnected(UNUSED sd_bus_message *message, void *userdata, UNU
 
         if (node->name) {
                 hirte_log_infof("Node '%s' disconnected", node->name);
+
+                /*
+                If the disconnected node is a providing one, we notified the requesting node about it.
+                Now clean up all pending proxy monitors related to that node.
+                */
+                ProxyMonitor *proxy_monitor = NULL;
+                ProxyMonitor *next_proxy_monitor = NULL;
+                LIST_FOREACH_SAFE(monitors, proxy_monitor, next_proxy_monitor, manager->proxy_monitors) {
+                        if (streq(proxy_monitor->providing_node->name, node->name) ||
+                            streq(proxy_monitor->requesting_node->name, node->name)) {
+                                proxy_monitor_close(proxy_monitor);
+                                manager_remove_proxy_monitor(manager, proxy_monitor);
+                        }
```

Comment:
I don't think this is the right way to handle this. We already sent the (virtual) unit events, and I think our handling of the termination of the proxy should be the same, independent of whether the events were virtual or real.

In other words, if we get *actual* reported events from the node that the service was stopped or unloaded, then we should report that to the requesting node and then stop the monitor. That could should also trigger from the virtual events, and thus there is no need for this special cleanup code.

Reply:
I guess we still want to do this if the requesting node disconnects though.

Reply:
Although if we drop the proxy request when the agent disconnects like this, we need the agent to realize that if the manager disconnects, then it needs to stop all outstanding proxies (both the initial call, and send a stop to all active proxies).